### ISSUE TITLE
Rework noaa.py to use the ghcnd_inventory data

### ISSUE THREAD

**Issue by rectheworld** [DataKind-DC/audubon-cbc#26]
# We need to Rematch to Station IDs (I know i'm sad too) 

## Why do we have to rematch?
Based off of the file closest_station.csv I was able to merge based on Lat Long with the Clenaed Data File to create the table 'usa_cleaned_matched' (can be viewed in our Google Project as a table or downloaded from our GCS) 

Using this file, I discovered a couple of things
1. Some of the stations (ie the ones that start with COOP) do no produce weather data (at least not in the bigquery-public-data.ghcn_d service 
2. Some of the matched stations are matched for the closest station id in 2019 -- this is not a viable matched station for circles from previous years. 
Example - For a count that occurred in  1960 that was matched to the station Id: US1IABH0003
There is a US1IABH0003 in 2019, but not in 1960. 

**In Short, Stations need to be matched by Year and Location**

I have collected the list of stations available from  bigquery-public-data.ghcn_d ghcn_inventory and filtered it to "uniquie" stations by year using this query 
```
SELECT DISTINCT(inventory.id), inventory.latitude, inventory.longitude, inventory.firstyear, inventory.lastyear
FROM `bigquery-public-data.ghcn_d.ghcnd_inventory` as inventory
```
The exported file I have saved here: https://drive.google.com/file/d/1MHoTkxRcMBM_hfAr1i4IyYvzlAFJjAbo/view?usp=sharing

**Comment by rectheworld** [comment]
After letting this file run for 48.2 hours and successfully getting the resulting file, I realized that I did a dumb thing. 

After getting the list of Closest NOAA stations, I realized I had no way to accurately merge the file by Station ID and year.  To keep from having to rerun the file again (another 42.2 hours) I cheated. 
Because I know what stations are capible of being matching because they wer already present in the file that took 42.2 to run. I created the file [station_inventory_cheater.csv](https://drive.google.com/open?id=19pzJZ7Hrnkr98SaNPp3KkrVCi_sdK0yy)

To make merging the matched data with the cbc data easier, I added the year and the row number to the output. I then reran the matching file match_with_noaa_stations.py (formally know as noaa.py) on just the smaller file.

**Comment by rectheworld** [comment]
The output file is [cbc_cleaned_usa_statid.csv](https://drive.google.com/open?id=1vhC4Q_QFfYrjKXJkChO4m8QPvxJdV8D_)

It is a merged file of Weather data and the closest NOAA station that was operating at the time of the Christmas Bird Count for that record.

**Comment by rectheworld** [comment]
Forgot to add the script that merges the noaa data with the cleaned cbc data. the file name is connecting_circles_with_pulled_station_id_data.ipynb

**Comment by rectheworld** [comment]
After discussing minh5, we decided that all files should be notebooks. So I am reopening so I can add the files in as notebooks
